insta/insta_api/urls.py

```diff
@@ -1,17 +1,21 @@
 from django.urls import path, include
 from .views import(
     PhotoListApiView,
+    PhotoDetailApiView,
     CommentListApiView,
     CommentDetailApiView,
+    CommentDetailApiView2,
     LikeListApiView,
     LikeDetailApiView,
     homePageView
 )
 
 urlpatterns = [
     path('photos', PhotoListApiView.as_view()),
+    path('photos/<int:photo_id>/', PhotoDetailApiView.as_view()),
     path('comments', CommentListApiView.as_view()),
     path('comments/<int:photo_id>/', CommentDetailApiView.as_view()),
+    path('comments/<int:comment_id>/comment/', CommentDetailApiView2.as_view()),
```

Comment:
I think it should be just `comments/<int:comment_id>` and the one with photo_id could be removed.

Reply:
Ok, but I have to ask how this particular endpoint should work. Here's my current approach:
path('comments', CommentListApiView.as_view()), -> handling GET (which returns all comments, no matter the photo, user etc.) and POST (posting new comment with given data)

path('comments/<int:photo_id>/', CommentDetailApiView.as_view()) -> handling only GET (which returns all comments under the photo - that means all comments with same photo_id)

path('comments/<int:comment_id>/comment/', CommentDetailApiView2.as_view()) -> handling GET (which returns one comment with given comment_id), PUT (which updates one comment with given comment_id), DELETE (which deletes one comment with given comment_id)

So now I want to know if my way of thinking is good, or something about comments is missing or maybe unnecessary. I'm writing this answer to provide some background, because at the moment if I just delete line 17 some functionality will be lost and I'm not sure is it intended.

Reply:
>handling only GET (which returns all comments under the photo - that means all comments with same photo_id)

If you want get all comments from single photo then with REST approach it should be like this `/photos/:id/comments` the main entity here is photo and comments are additional resource :D

>comments/int:comment_id/comment/

This pattern is not correct with rest approach :D all actions around comments should be defined like this
GET /comments -> list all of them (not needed in your case, app will load comments per post, not all of them)
GET /comments/:id -> get single comment
POST /comments -> create new comment (not needed in your case as it should be like this `POST /photos/:id/comments`
PATCH /comments/:id -> update single comment
DELETE /comments/:id -> remove single comment :D